server/app/com/xsn/explorer/tasks/MasternodeSynchronizerTask.scala

```diff
@@ -0,0 +1,72 @@
+package com.xsn.explorer.tasks
+
+import akka.actor.{Actor, ActorSystem, Props}
+import com.xsn.explorer.config.MasternodeSynchronizerConfig
+import com.xsn.explorer.models._
+import com.xsn.explorer.services.{XSNService}
+import javax.inject.Inject
+import org.scalactic.{Bad, Good}
+import org.slf4j.LoggerFactory
+import com.xsn.explorer.models.values.IPAddress
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+class MasternodeSynchronizerActor extends Actor {
+  import context._
+
+  def receive: Receive = {
+    val initialMasternodes = List[rpc.Masternode]()
+
+    behavior(initialMasternodes)
+  }
+
+  private def behavior(masternodes: List[rpc.Masternode]): Receive = {
+    case MasternodeSynchronizerActor.UpdateMasternodes(newMasternodes) =>
+      become(behavior(newMasternodes))
+    case MasternodeSynchronizerActor.GetMasternode(ipAddress) =>
+      sender() ! masternodes.find(x => x.ip.split(":")(0) == ipAddress.string)
```

Comment:
`x.ip.split(":")(0)` could throw exceptions, instead `x.ip.split(":").headOption.contains(ipAddress.string))` would be safe`, also, I'd return a typed message like `MasternodeSynchronizerActor.GotMasternode(maybe)` where `maybe` is `Option[rpc.Masternode]`